src/parse/parser.rs

```diff
@@ -1653,27 +1644,21 @@ where
         match Help::new(HelpWriter::Buffer(&mut c), self, use_long).write_help() {
             Err(e) => e,
             _ => ClapError {
-                cause: String::new(),
                 message: c,
                 kind: ErrorKind::HelpDisplayed,
-                info: None,
             },
         }
     }
 
     fn version_err(&self, use_long: bool) -> ClapError {
         debug!("Parser::version_err");
 
-        let mut c = Colorizer::new(false, self.app.color());
-
-        match self.print_version(&mut c, use_long) {
-            Err(e) => e,
-            _ => ClapError {
-                cause: String::new(),
-                message: c,
-                kind: ErrorKind::VersionDisplayed,
-                info: None,
-            },
+        let msg = self.app._render_version(use_long);
+        let mut c = Colorizer::new(false, ColorChoice::Never);
```

Comment:
```suggestion
        let mut c = Colorizer::new(false, self.app.color());
```

Reply:
We never color the version, so it doesn't matter.

Reply:
We never color the version, so it doesn't matter.

Reply:
Still, if we decide to color in the future. Always passing app's color is better when constructing colorizer.